app/models/ncr/proposal_form.rb

```diff
@@ -59,23 +59,49 @@ def create_cart
         name: self.description
       )
       if cart.save
-        cart.set_props(
-          origin: self.origin,
-          amount: self.amount,
-          expense_type: self.expense_type,
-          vendor: self.vendor,
-          not_to_exceed: self.not_to_exceed,
-          building_number: self.building_number,
-          rwa_number: self.rwa_number,
-          office: self.office
-        )
-        cart.set_requester(self.requester)
+        self.set_props_on(cart)
         self.approver_emails.each do |email|
           cart.add_approver(email)
         end
+        Dispatcher.deliver_new_cart_emails(cart)
       end
 
       cart
     end
+
+    def update_cart(cart)
+      cart.name = self.description
+      if cart.save
+        cart.approver_approvals.destroy_all
+        cart.clear_props!
```

Comment:
This is meant to remove any weird versioning issues -- e.g. fields that were present in a previous version of the form but are not anymore. It would also remove other properties set on the cart, if we began collecting more meta data, which would probably be bad.

Maybe a list of fields (which only grows) to remove could account for both types of bugs?